enhancements/installer/component-selection.md

```diff
@@ -0,0 +1,381 @@
+other component that assumes there is always an imageregistry or assumes the
+presence of some CRD api that is installed with the imageregistry to break).
+
+There was some discussion about the pros/cons of allowing each component to be enabled/disabled independent
+of that component explicitly opting into a particular (presumably well tested) configuration/topology
+[here](https://github.com/openshift/enhancements/pull/200#discussion_r375837903).  The position of this EP is that
+we should only recommend the exclusion of fully independent "capability" components that are not depended on by
+other components.  Further the assumption is that it will be reasonable to tell a customer who disabled
+something and ended up with a non-functional cluster that their chosen exclusions are simply not supported
+currently, or that they must accept the degraded functionality caused by the missing dependency, if they
+intend to keep it disabled.
+
+Since the only components/resources that can be filtered out of the installation are ones that are explicilty
+annotated with `capability.openshift.io/$component`, end-users will not be able to use this mechanism to filter
+components/resources that we did not intend for them to be able to filter out.
+
+## Design Details
+
+### Answered Questions
+
+
+1. Do we want to constrain this functionality to turning off individual capabilities?  We could
+also use it to
+  a) turn on/off groups of components as defined by "solutions" (e.g. a "headless" solution
+  which might turn off the console but also some other capabilities).  This is what CLUSTER_PROFILES
+  sort of enable, but there seems to be reluctance to expand the cluster profile use case to include
+  these sorts of things.
+  b) enable/disable specific configurations/topologies such as "HA", where components could contribute multiple
+  deployment definitions for different configurations and then the installer/CVO would select the correct
+  one based on the chosen install configuration (HA vs single node) instead of having components read/reconcile
+  the infrastructure resource.
+
+Current plan is to constrain this functionality to capability level controls.
+
+2. How does the admin enable a component post-install if they change their mind about what components
+they want enabled?  Do we need/want to allow this?
+
+Turning on a component later is relatively easy (we expose a config resource for the CVO that defines
+the filter, we allow the user to remove items from the filter, the CVO will apply the previously
+filtered resources during the next reconciliation).
+
+Turning off a component later is more problematic because
+a) The CVO doesn't delete resources today, so that would be a new thing to teach it to do.
+b) Just deleting the resources for the component isn't sufficient, as the component also needs to clean
+itself up in case it created any additional resources on the cluster or contributed any configuration.
```

Comment:
This is an issue, but it's not insurmountable.  Certainly a blanket "not supported" means we don't have to do the per-component analysis to decide how much hoop-jumping removing each particular component would be.  Also on this front, uninstalling OLM-installed operators today can require [some admin hoop-jumping to remove dangling resources][1].  So worst-case situations here are:

1. We doc some manual steps for each component that is uninstallable, or
2. Components that are hard to auto-remove because of dangling resources, but where we don't want to doc admin steps, don't actually get uninstalled, but do grow something like [registry's `managementState: Removed`][2], where a small shim operator sticks around and uses its robot brains and some residual creds to automatically clean out the heavy resources.

But if there's not much pressure to support day-2 removal, we can just say "folks don't really need this", and punt on a detailed breakdown of the hills we'd need to climb to support it.

[1]: https://docs.openshift.com/container-platform/4.8/service_mesh/v1x/removing-ossm.html#ossm-remove-cleanup-1x_removing-ossm-v1x
[2]: https://docs.openshift.com/container-platform/4.8/registry/configuring-registry-operator.html#registry-operator-configuration-resource-overview_configuring-registry-operator